rnwinrt/rnwinrt/react/strings/base.cpp

```diff
@@ -205,22 +205,29 @@ jsi::Value static_enum_data::create(jsi::Runtime& runtime) const
     return jsi::Value(runtime, jsi::Object::createFromHostObject(runtime, std::make_shared<projected_enum>(this)));
 }
 
-jsi::Value static_enum_data::get_value(std::string_view valueName) const
+jsi::Value static_enum_data::get_value(jsi::Runtime& runtime, std::string_view valueName) const
 {
     // TODO: It would also be rather simple to ensure that the array is sorted and do a binary search, however the
     // number of enum elements is typically pretty small, so that may actually be harmful
     auto itr = std::find_if(values.begin(), values.end(), [&](auto& mapping) { return mapping.name == valueName; });
-    if (itr == values.end())
+    if (itr != values.end())
     {
-        return jsi::Value::undefined();
+        return jsi::Value(itr->value);
     }
 
-    return jsi::Value(itr->value);
+    // Look for a matching value (reverse mapping)
+    itr = std::find_if(values.begin(), values.end(), [&](auto& mapping) { return std::to_string(static_cast<int32_t>(mapping.value)) == valueName; });
```

Comment:
I'm okay with either: (1) storing the string in the metadata (and therefore not needing to call `std::to_string`), or (2) converting the name to a `double`. We should do one or the other.

Reply:
Option 2 goes against the way JavaScript works. If we go with option 1, how do you suggest settings those strings in the metadata? It seems that any call to `std::string` will prevent the function from being a `constexpr`.

Reply:
> how do you suggest settings those strings in the metadata? It seems that any call to std::string will prevent the function from being a constexpr.

By not using `std::string`.
```c++
        struct value_mapping
        {
            std::string_view name;
            double value;
            std::string_view value_string;
        };
```
And then update the [codegen](https://github.com/microsoft/react-native-winrt/blob/9b8b4db28e4dcbd72c405e2e1603116ec6c20a98/rnwinrt/rnwinrt/react/ReactFileGenerator.cpp#L400) to include the value using a string literal

Reply:
Okay. I think I got it now. Let me know what you think.